### DISCUSSION TITLE
Pyodide-build: FileNotFoundError: Could not find Pyodide root directory. If you are not in the Pyodide directory, set `PYODIDE_ROOT=<pyodide-root-directory>`.

### DISCUSSION THREAD

**Discussion by s3131212** [pyodide/pyodide#3669] [Q&A] [answered]
I installed pyodide-build following the tutorial on the [documentation](https://pyodide.org/en/stable/development/building-and-testing-packages.html).  

After installing via `pip3 install pyodide-build`, the command `pyodide` is not found, and `pyodide-build` and `python3 -m pyodide_build` are presented.

However, running both command result in the following error:
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/user/.local/lib/python3.10/site-packages/pyodide_build/__main__.py", line 50, in <module>
    main()
  File "/home/user/.local/lib/python3.10/site-packages/pyodide_build/__main__.py", line 37, in main
    init_environment()
  File "/home/user/.local/lib/python3.10/site-packages/pyodide_build/common.py", line 344, in init_environment
    os.environ["PYODIDE_ROOT"] = str(search_pyodide_root(os.getcwd()))
  File "/home/user/.local/lib/python3.10/site-packages/pyodide_build/common.py", line 328, in search_pyodide_root
    raise FileNotFoundError(
FileNotFoundError: Could not find Pyodide root directory. If you are not in the Pyodide directory, set `PYODIDE_ROOT=<pyodide-root-directory>`.
```

I am not sure what `PYODIDE_ROOT` means. I set it to the working directory, i.e., the directory of my project, and the installation path of pyodide-build. Both do not help.  

I installed pyodide-build before (when Pyodide was still in 0.21.3) and everything worked fine then. I tried to upgrade to the newest version (via `pip3 install --upgrade`) and it no longer works. Reinstalling the package completely also does not help.

System info:
```
Manjaro (Linux kernel: 6.0.19)
Python 3.10.9
pyodide-cli==0.2.1
pyodide-build==0.22.1
```

Any help is appreciated!

**Comment by hoodmane** [comment]
Thanks for the report @s3131212! We've just got three bug reports about this in the last few days so there must be a regression here. It's odd that we're seeing these now since our last release was 7 weeks ago. Anyways I'll try to investigate as soon as I get an opportunity.

**Reply by hoodmane** [reply]
For now, try
```shell
wget https://github.com/pyodide/pyodide/releases/download/0.22.1/xbuildenv-0.22.1.tar.bz2
tar -xf xbuildenv-0.22.1.tar.bz2
export PYODIDE_ROOT=$(pwd)/folder_just_extracted
```
I think it should work then???

**Comment by hoodmane** [comment, marked as answer]
I don't reproduce the problem. 
```console
$ python3.10 -m venv .venv
$ source .venv/bin/activate
$ pip install pyodide-build
Collecting pyodide-build
  Downloading pyodide_build-0.22.1-py3-none-any.whl (60 kB)
[...]
$ which pyodide
/home/ubuntu/numpy/.venv/bin/pyodide
```
so the `pyodide` cli did get installed.
Then:
```console
$ pyodide build
Downloading xbuild environment
Installing xbuild environment
RuntimeError: No Emscripten compiler found. Need Emscripten version 3.1.27
```
Okay I need to install Emscripten:
```console
$ gh repo clone emscripten-core/emsdk
Cloning into 'emsdk'...
[...]
$ emsdk/emsdk install 3.1.27
Resolving SDK version '3.1.27' to 'sdk-releases-48ce0b44015d0182fc8c27aa9fbc0a4474b55982-64bit'
[...]
$ emsdk/emsdk activate 3.1.27
Setting the following tools as active:
[...]
$ source emsdk/emsdk_env.sh
Setting up EMSDK environment (suppress these messages with EMSDK_QUIET=1)
[...]
$ pyodide build
* Creating virtualenv isolated environment...
[...]
[...]
```
and it successfully builds the wheel (I used numpy for this test case).

**Reply by hoodmane** [reply]
So I wonder why you see different results... We'll have to investigate further.

**Reply by s3131212** [reply]
Creating an venv works. Thanks for your help!

Not sure if this is expected. I used the exact same commands you wrote. The Pyodide seems to request the wrong Emscripten version. According to my unreliable memory, Emscripten 3.1.14 was used in Pyodide v0.21.3, which was the Pyodide version I installed previously.

```
$ pyodide --version
Pyodide CLI Version: 0.2.3
$ pyodide config list
emscripten_version=3.1.14
python_version=3.10.2
$ pyodide build
[...]
RuntimeError: Incorrect Emscripten version 3.1.27. Need Emscripten version 3.1.14
```

I have to assign an environment variable to change the Emscripten version to the one required by the latest Pyodide release.
```
$ PYODIDE_EMSCRIPTEN_VERSION=3.1.27 pyodide build
```

Then everything works as expected. The built wheel runs perfectly using the latest Pyodide release.

I've uninstalled pyodide-build and pyodide-cli using `pip uninstall` before. Not sure why this happens...

**Reply by hoodmane** [reply]
You need to remove the old `.pyodide-xbuildenv`.

**Reply by s3131212** [reply]
Sorry did not notice the config file. It works perfectly now. Thanks a lot!

**Comment by ryanking13** [comment]
I think #3544 would hopefully fix the bug here, thanks for the report!
